frontend/src/lib/components/DebugNotice.tsx

```diff
@@ -1,17 +1,40 @@
 import { IconCode, IconX } from '@posthog/icons'
 import { useValues } from 'kea'
-import { IconBranch } from 'lib/lemon-ui/icons'
+import { IconBranch, IconOpenInNew } from 'lib/lemon-ui/icons'
 import { LemonButton } from 'lib/lemon-ui/LemonButton'
-import { useEffect, useState } from 'react'
+import { useEffect, useMemo, useState } from 'react'
 
 import { NavbarButton } from '~/layout/navigation-3000/components/NavbarButton'
 import { navigation3000Logic } from '~/layout/navigation-3000/navigationLogic'
 
 export function DebugNotice(): JSX.Element | null {
     const [debugInfo, setDebugInfo] = useState<{ branch: string; revision: string } | undefined>()
     const [noticeHidden, setNoticeHidden] = useState(false)
+    const [isPort8000, setIsPort8000] = useState(false)
+
     const { isNavCollapsed } = useValues(navigation3000Logic)
 
+    useEffect(() => {
+        const isLocal = ['127.0.0.1', '0.0.0.0', 'localhost'].includes(window.location.hostname)
+        const port = window.location.port
+
+        if (isLocal && port === '8000') {
+            setIsPort8000(true)
+        }
```

Comment:
ladies and gentlemen, here I am answering the question, how over-engineered can you make the check for a URL